### DISCUSSION TITLE
Cannot start xrdp service

### DISCUSSION THREAD

**Discussion by CoolCat64** [microsoft/WSL#6238] [General]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19042.630]
Your Distribution version: lsb_release -a
Distributor ID: Kali
Description:    Kali GNU/Linux Rolling
Release:        2020.4
Codename:       kali-rolling
Whether the issue is on WSL 2 and/or WSL 1: 
Linux version 4.19.128-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Tue Jun 23 12:58:10 UTC 2020
```

# Steps to reproduce

PS D:\> wsl -d kali-linux -u root -e bash -c "/etc/init.d/xrdp start"
Starting Remote Desktop Protocol server: xrdp-sesman xrdp.
PS D:\> wsl -d kali-linux -u root -e bash -c "/etc/init.d/xrdp status"
xrdp-sesman is running.
xrdp is not running ... failed!

Running mstsc.exe on localhost times out.

**WSL logs**: https://aka.ms/AAabf4a

#  Expected behavior

I expect the output of /etc/init.d/xrdp status to be:
xrdp-sesman is running.
xrdp is running.

I expect to be able to connect to the xrdp via mstsc.exe on localhost

# Actual behavior

xrdp never starts. I cannot make an RDP connection on localhost.

**Comment by Myron-S** [comment]
I've had this issue recently then all of a sudden it started working, but I'm wondering if it's something I did.

Kernel I'm using:
`Linux version 5.4.72-microsoft-standard-WSL2 (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Oct 28 23:40:43 UTC 2020`

Second was is that I slightly altered %USERPROFILE%\.wslconfig to this:
```
[wsl2]
memory = 4GB
localhostForwarding = true

```
The odd thing here is I think the RDP client started to connect to XRDP when I put a space to the left and right of the equals sign.

I'm also starting systemd with genie which starts xrdp and quite a number of other services.  [Runs systemd in a bottle as PID 1.]

**Comment by zflash123** [comment]
I have the same problem just now. What I do to fix this issue is to check is there "xrdp.ini" file on "/etc/xrdp/" directory
First I run 
`cd /etc/xrdp/`
And then run
`ls | grep xrdp.ini`
If above command not return anything, it is because your "xrdp.ini" file is missing.
So you can add your "xrdp.ini" file your self.

What you need to do is just copy the "xrdp.ini.original" file from "/etc/xrdp/" directory and rename the file as "xrdp.ini". To do this, just run:
`cp xrdp.ini.original xrdp.ini`
Restart your xrdp service:
`service xrdp restart`

And it is successful

**Reply by evilat** [reply]
thank you very much brother love you you are my brooooooo

**Reply by zflash123** [reply]
no problem, happy to help

**Comment by anushhkumar** [comment]
I'm also starting systemd with genie which starts xrdp and quite a number of other services. [Runs [systemd](https://sourceforge.net/p/seb/discussion/844844/thread/4e9583ee5c/) in a bottle as PID 1
